tests/nat/builder/test_per_user_builder.py

```
@@ -0,0 +1,1225 @@
+
+        yield FunctionInfo.from_fn(_impl)
+
+    config = Config(functions={
+        "per_user_fn": PerUserFunctionConfig(),
+        "bad_shared_fn": SharedDependentFnConfig(per_user_fn_name="per_user_fn"),
+        WORKFLOW_COMPONENT_NAME: SharedWorkflowConfig(),
+    },
+                    workflow=SharedWorkflowConfig())
+
+    # Building should fail - either with validation error or when trying to get per-user function
+    with pytest.raises(ValueError, match="(depends on per-user function|Function `per_user_fn` not found)"):
```

Ruff (0.14.7)

Pattern passed to `match=` contains metacharacters but is neither escaped nor raw

(RUF043)